pkg/util/clusterdata/clusterdata.go

```diff
@@ -117,6 +121,26 @@ out:
 	}
 }
 
+// isValidProvisioningState checks whether or not it is ok to run enrichment
```

Comment:
Should we run this only if the cluster is in Successful state? It can be failed due to `Updating` and `AdminUpdating` too. And we might be safer doing this only on "Successful cluster". On top of that logic would become much cleaner and transparent in API model:

```
// IsTerminal returns true if state is Terminal
func (t ProvisioningState) IsSucceeded() bool {
	return ProvisioningStateSucceeded == t
}
```

And when customer comes to us, we have one and single answer - cluster if in a failed state. As now it gets a little bit confusing when it runs and when it does not.

Reply:
If decide to run enrichment only in `ProvisioningStateSucceeded` - we will be reporting invalid state of worker profiles, version, etc to a customer during the update.

Also in the failed `AdminUpdating` and `Updating` states a customer can have a healthy cluster and it will be confusing that the cluster works fine, but RP does not report worker profiles, for example.

Reply:
I get the consequence, but Im thinking of all the confusion it might create why sometimes it works and sometimes it does not... Might be a small price to pay

Reply:
On the other hand, 
I scale up cluster. And It failed. I want to check what is the current state, how much it managed to scale before failing. 
Complex...

Reply:
I am +1 on how this PR is implemented currently.  In the future I want us to cache the last enriched output, that will help with the Deleting/Failed-Deleting phases.